portfolio/src/main/java/com/google/sps/servlets/LoadForm.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import com.google.protobuf.Message;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing tasks. */
+@WebServlet("/list-messages")
+public class LoadForm extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+   
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query =
+        Query.newEntityQueryBuilder().setKind("Task").setOrderBy(OrderBy.desc("name")).build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    List<String []> messages = new ArrayList<>();
+    while (results.hasNext()) {
+      Entity entity = results.next();
+      String name = entity.getString("name");
+      String email = entity.getString("email");
```

Comment:
I am going to enforce the suggestion in the walkthrough about `You should also consider hiding personal data like email addresses.` by not displaying the email on the webpage

Unfortunately, not clearly explained in the walkthrough, but the main concern is once someone uses your website, their email is exposed to the internet without their consent. And they have no way to remove it without contacting you, which isn't great for the privacy of their data.

Having it in Datastore is different because it is protected by authorization requirements and only you can access the emails there.